Sources/ApodiniDeploy/ApodiniDeployInterfaceExporter.swift

```diff
@@ -0,0 +1,250 @@
+                    )
+                }
+                let runtimeSupport = try DPRSType.init(deployedSystem: deployedSystem, currentNodeId: currentNodeId)
+                self.deploymentProviderRuntime = runtimeSupport
+                try runtimeSupport.configure(app)
+            } catch {
+                throw ApodiniDeployError(message: "Unable to launch with custom config: \(error)")
+            }
+            
+        default:
+            break
+        }
+    }
+    
+    
+    func getEndpoint<H: IdentifiableHandler>(withIdentifier identifier: H.HandlerIdentifier, ofType _: H.Type) -> Endpoint<H>? {
+        collectedEndpoints.first { $0.endpoint.identifier == identifier }?.endpoint as? Endpoint<H>
+    }
+    
+    
+    public func retrieveParameter<Type: Codable>(_ endpointParameter: EndpointParameter<Type>, for request: ExporterRequest) throws -> Type?? {
+        guard let paramValueContainer = request.getValueOfCollectedParameter(for: endpointParameter) else {
+            return Optional<Type?>.none // this should be a "top-level" nil value (ie `.none` instead of `.some(.none)`)
+        }
+        
+        if endpointParameter.nilIsValidValue {
+            switch paramValueContainer {
+            case .value(let value):
+                if let value: Type? = dynamicCast(value, to: Type?.self) {
+                    return .some(value)
+                }
+            case .encoded(let data):
+                return try .some(JSONDecoder().decode(Type?.self, from: data))
+            }
+        } else {
+            switch paramValueContainer {
+            case .value(let value):
+                if let value = value as? Type {
+                    return .some(.some(value))
+                }
+            case .encoded(let data):
+                return try .some(.some(JSONDecoder().decode(Type.self, from: data)))
+            }
+        }
+        throw ApodiniDeployError(
+            message: "Unable to cast parameter value (\(paramValueContainer)) to expected type '\(Type.self)'"
+        )
+    }
+}
+
+
+// MARK: ApodiniDeployInterfaceExporter.ExporterRequest
+
+extension ApodiniDeployInterfaceExporter {
+    public struct ExporterRequest: Apodini.ExporterRequest {
+        enum Param {
+            case value(Any)    // the value, as-is
+            case encoded(Data) // the value, encoded
+        }
+        
+        private let parameterValues: [String: Param] // key: stable endpoint param identity
+        
+        init<H: Handler>(endpoint: Endpoint<H>, collectedParameters: [CollectedParameter<H>]) {
+            parameterValues = .init(uniqueKeysWithValues: collectedParameters.map { param -> (String, Param) in
+                let paramId = (endpoint.handler[keyPath: param.handlerKeyPath] as! AnyParameterID).value
+                let endpointParam = endpoint.parameters.first { $0.id == paramId }!
```

Comment:
Nope, this works the other way around: we're mapping the set of collected parameters (i.e. the parameters passed to the remote invocation API) to their corresponding `EndpointParameter`s.
Because the parameter objects (think of these as key-value pairs mapping a parameter key path to the value passed for that parameter) you pass to the remote handler invocation API can only be you can only constructed by referencing an `@Parameter`'s key path, we know that every collected parameter has some corresponding `@Parameter`.
And because every `@Parameter` has a corresponding `EndpointParameter`, we can guarantee that this mapping from collectedParameters to EndpointParameters will always succeed.

summary: we're mapping from a subset to a superset, therefore we will always find the matching EndpointParameter and the force unwrap should never fail